lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -72,6 +73,22 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+        user_id = usage_info.user.id if usage_info.user else None
+
         for block_key in block_structure.post_order_traversal():
             if block_structure.get_xblock_field(block_key, 'category') not in self.requested_student_view_data:
                 block_structure.remove_transformer_block_field(block_key, self, self.STUDENT_VIEW_DATA)
+            student_view_data = block_structure.get_transformer_block_field(
+                block_key,
+                self,
+                self.STUDENT_VIEW_DATA,
+            )
+
+            if student_view_data and user_id and 'html' in student_view_data:
+                student_view_data['html'] = student_view_data['html'].replace('%%USER_ID%%', str(user_id))
+                block_structure.set_transformer_block_field(
+                    block_key,
+                    self,
+                    self.STUDENT_VIEW_DATA,
+                    student_view_data
+                )
```

Comment:
I wonder if it could break any assumptions among API consumers if `student_view_data` is in this one case specific to the current student and not the same for all students.

Reply:
I wouldn't think it would be an issue - since API consumers do expect to receive different block-structures for different users - given differing (staff/group/etc) accesses.  If you weren't checking for `usage_info.user`, then that would be a concern since API consumers that don't pass in a `username` expect to receive the user-agnostic values of the block-structure.